doozer/doozerlib/cli/images_konflux.py

```diff
@@ -257,39 +259,49 @@ async def get_operator_builds(self):
-                names=[metadata.distgit_key for metadata in operator_metas],
-                group=runtime.group,
-                assembly=runtime.assembly,
-                artifact_type=ArtifactType.IMAGE,
-                engine=Engine.KONFLUX,
-                strict=True,
-            )
+            records = await asyncio.gather(*(metadata.get_latest_build() for metadata in operator_metas))
+            not_found = [metadata.distgit_key for metadata, record in zip(operator_metas, records) if record is None]
+            if not_found:
+                raise IOError(f"Couldn't find build records for {not_found}")
             for metadata, record in zip(operator_metas, records):
                 assert record is not None and isinstance(record, KonfluxBuildRecord)
                 dgk_records[metadata.distgit_key] = record
         return dgk_records
 
+    async def _get_bundle_build_for(self, operator_build: KonfluxBuildRecord, strict: bool = True) -> Optional[KonfluxBundleBuildRecord]:
+        """ Get bundle build record for the given operator build.
+
+        :param operator_build: Operator build record.
+        :return: Bundle build record.
+        """
+        bundle_name = f"{operator_build.name}-bundle"
+        LOGGER.info("Fetching bundle build for %s from Konflux DB...", operator_build.nvr)
+        where = {
+            "name": bundle_name,
+            "group": self.runtime.group,
+            "assembly": self.runtime.assembly,
+            "operator_nvr": operator_build.nvr,
+            "outcome": str(KonfluxBuildOutcome.SUCCESS),
+        }
+        bundle_build = await anext(self._db_for_bundles.search_builds_by_fields(where=where, limit=1), None)
+        if not bundle_build:
+            if strict:
+                raise IOError(f"Bundle build not found for {operator_build.name}. Please build the bundle first.")
+            return None
+        assert isinstance(bundle_build, KonfluxBundleBuildRecord)
+        return bundle_build
+
     async def _rebase_and_build(
             self,
             rebaser: KonfluxOlmBundleRebaser,
             builder: KonfluxOlmBundleBuilder,
-            db_for_bundles: KonfluxDb,
             image_meta: ImageMetadata,
             operator_build: KonfluxBuildRecord):
         logger = LOGGER.getChild(f"[{image_meta.distgit_key}]")
-        runtime = self.runtime
-        assembly = runtime.assembly
         input_release = self.release
         if not self.force or not input_release:
             logger.info("Checking if a previous bundle build exists...")
-            bundle_build = await db_for_bundles.get_latest_build(
-                name=image_meta.get_olm_bundle_short_name(),
-                group=runtime.group,
-                assembly=assembly,
-                engine=Engine.KONFLUX,
-                strict=False,
-            )
+            bundle_build = await self._get_bundle_build_for(operator_build, strict=False)
```

Comment:
We should find the latest bundle for the assembly *and the specific NVR*. The currently implementation misses the second condition.